shanoir-ng-import/src/main/java/org/shanoir/ng/importer/dicom/query/QueryPACSService.java

```diff
@@ -409,38 +439,75 @@ private void queryInstances(DicomNode calling, DicomNode called, Serie serie, St
 	
 	/**
 	 * This method does a C-FIND query and returns the results.
-	 * @param params
+	 * 
+	 * The state of each c-find query is a local attribute of the method.
+	 * So, when e.g. on the server 3 users call in parallel queryCFind,
+	 * each one has its own DimseRSPHandler and its own state, so this
+	 * might work, in case the association is not caching aspects.
+	 * 
+	 * @param keys
 	 * @param level
-	 * @param calling
-	 * @param called
 	 * @return
 	 */
-	private List<Attributes> queryCFIND(DicomParam[] params, QueryRetrieveLevel level, final DicomNode calling, final DicomNode called) {
-		AdvancedParams options = new AdvancedParams();
+	private List<Attributes> queryCFind(DicomParam[] keys, QueryRetrieveLevel level) {
+		String cuid = null;
 		if (level.equals(QueryRetrieveLevel.PATIENT)) {
-			options.setInformationModel(InformationModel.PatientRoot);
+			cuid = UID.PatientRootQueryRetrieveInformationModelFind;
 		} else if (level.equals(QueryRetrieveLevel.STUDY)) {
-			options.setInformationModel(InformationModel.StudyRoot);
+			cuid = UID.StudyRootQueryRetrieveInformationModelFind;
 		} else if (level.equals(QueryRetrieveLevel.SERIES)) {
-			options.setInformationModel(InformationModel.StudyRoot);
+			cuid = UID.StudyRootQueryRetrieveInformationModelFind;
 		} else if (level.equals(QueryRetrieveLevel.IMAGE)) {
-			options.setInformationModel(InformationModel.StudyRoot);
+			cuid = UID.StudyRootQueryRetrieveInformationModelFind;
 		}
-		logQuery(params, options);
-		DicomState state = CFind.process(options, calling, called, 0, level, params);
-		return state.getDicomRSP();
-	}
-
-	/**
-	 * This method logs the params and options of the PACS query.
-	 * @param params
-	 * @param options
-	 */
-	private void logQuery(DicomParam[] params, AdvancedParams options) {
-		LOG.info("Calling PACS, C-FIND with level: {} and params:", options.getInformationModel());
-		for (int i = 0; i < params.length; i++) {
-			LOG.info("Tag: {}, Value: {}", params[i].getTagName(), Arrays.toString(params[i].getValues()));
+		DicomState state = new DicomState(new DicomProgress());
```

Comment:
Use this DicomProgress ? (is it necessary otherwise ?)

Reply:
I took it from dcm4che3 example, there might be a reason why the start the state like this. I did not check.